src/server/scripts/Outland/zone_shadowmoon_valley.cpp

```diff
@@ -1429,7 +1453,74 @@ class npc_enraged_spirit : public CreatureScript
 
         void Reset() override { }
 
-        void EnterCombat(Unit* /*who*/) override { }
+        void EnterCombat(Unit* /*who*/) override
+        {
+            switch (me->GetEntry())
+            {
+                case NPC_ENRAGED_WATER_SPIRIT:
+                    _events.ScheduleEvent(EVENT_ENRAGED_WATER_SPIRIT, Seconds(0), Seconds(1));
+                    break;
+                case NPC_ENRAGED_FIRE_SPIRIT:
+                    _events.ScheduleEvent(EVENT_ENRAGED_FIRE_SPIRIT, Seconds(0), Seconds(1));
+                    break;
+                case NPC_ENRAGED_EARTH_SPIRIT:
+                    _events.ScheduleEvent(EVENT_ENRAGED_EARTH_SPIRIT, Seconds(0), Seconds(1));
+                    break;
+                case NPC_ENRAGED_AIR_SPIRIT:
+                    _events.ScheduleEvent(EVENT_ENRAGED_AIR_SPIRIT_CHAIN_LIGHTNING, Seconds(0), Seconds(1));
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+        _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_ENRAGED_WATER_SPIRIT:
+                        if (UpdateVictim())
+                            DoCastVictim(SPELL_STORMBOLT);
+                        _events.Repeat(Seconds(6), Seconds(9));
+                        break;
+                    case EVENT_ENRAGED_FIRE_SPIRIT:
+                        if (UpdateVictim())
+                            DoCastVictim(SPELL_FEL_FIREBALL);
+                        _events.Repeat(Seconds(7), Seconds(9));
+                        break;
+                    case EVENT_ENRAGED_EARTH_SPIRIT:
+                        if (UpdateVictim())
+                            DoCastVictim(SPELL_FIERY_BOULDER);
+                        _events.Repeat(Seconds(2), Seconds(3));
+                        break;
+                    case EVENT_ENRAGED_AIR_SPIRIT_CHAIN_LIGHTNING:
+                        if (UpdateVictim())
+                            DoCastVictim(SPELL_CHAIN_LIGHTNING);
+                        _events.ScheduleEvent(EVENT_ENRAGED_AIR_SPIRIT_HURRICANE, Seconds(3), Seconds(5));
+                        break;
+                    case EVENT_ENRAGED_AIR_SPIRIT_HURRICANE:
+                        if (UpdateVictim())
+                            DoCastVictim(SPELL_HURRICANE);
+                        _events.ScheduleEvent(EVENT_ENRAGED_AIR_SPIRIT_CHAIN_LIGHTNING, Seconds(10), Seconds(13));
+                        break;
+                    default:
+                        break;
+                }
+            }
+
+		if (!UpdateVictim())
+			return;
+
+		if (me->GetEntry() == NPC_ENRAGED_FIRE_SPIRIT || me->GetEntry() == NPC_ENRAGED_AIR_SPIRIT)
+			if (HealthBelowPct(35) && !me->GetAura(SPELL_ENRAGE))
```

Comment:
should be: !me->HasAura(SPELL_ENRAGE), you dont need get this aura...

Reply:
If i don't get this aura then the npc is just going to be casting enrage all the time when below 35% health

Reply:
I even tried using a boolean to validate the enrage but if the npc is channeling for example hurricane, then he turns it to true and doesn't cast the spell. That's the only reason i chose that logic.

Reply:
try: 
```cpp
if (HealthBelowPct(35) && !me->HasAura(SPELL_ENRAGE))
    DoCastSelf(SPELL_ENRAGE, true);
```